src/main/java/com/uid2/shared/secure/AzureCCCoreAttestationService.java

```diff
@@ -43,6 +46,11 @@ public void attest(byte[] attestationRequest, byte[] publicKey, Handler<AsyncRes
             log.debug("Validating signature...");
             var tokenPayload = tokenSignatureValidator.validate(tokenString);
 
+            if (tokenPayload != null && !UrlEquivalenceValidator.areUrlsEquivalent(tokenPayload.getRuntimeData().getAttestationUrl(), this.allowedAttestationUrl, log)) {
```

Comment:
This will fail attestation if the attestation url is not provided

Reply:
I don't think this is going to work as it stands. The attestationUrl in the userData is encoded and needs to be decoded to a string using something like this: new String(userData, StandardCharsets.UTF_8)
But that should be on the RuntimeData class